lib/triggerFactory.js

```diff
@@ -28,6 +28,43 @@ class TriggerFactory extends BaseFactory {
         return new Trigger(config);
     }
 
+    /**
+     * Get all triggers related to a pipeline
+     * @param  {Object} config
+     * @param  {String} config.pipelineId   pipelineId
+     * @return {Object}                     Returns object with dest triggers split by src jobs
+     */
+    getTriggers({ pipelineId }) {
+        // eslint-disable-next-line global-require
+        const PipelineFactory = require('./pipelineFactory');
+        const pipelineFactory = PipelineFactory.getInstance();
+        const result = [];
+
+        return pipelineFactory.get(pipelineId)
+            .then((pipeline) => {
+                if (pipeline) {
+                    // Get pipeline job names
+                    return pipeline.getJobs()
+                        .then(jobs => Promise.all(
+                            jobs.map(j =>
```

Comment:
do we need to filter out PR jobs ?

Reply:
if PRs never have external triggers, yes

Reply:
updated

Reply:
that seems to be the ask here https://github.com/screwdriver-cd/screwdriver/issues/1606 but we can get there much later, it needs a lot of design changes,

Reply:
yeah I think that can be addressed at a later time